test/mpi/init/session_psets.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include "mpi.h"
+#include <stdio.h>
+#include "mpitest.h"
+
+/* This is adapted example 10.9 from MPI Standard 4.0 */
+
+static char use_pset_name[] = "SELF";
+
+int main(int argc, char *argv[])
+{
+    int errs = 0;
+    int i, n_psets, psetlen, rc, ret;
+    int valuelen;
+    int flag = 0;
+    char *pset_name = NULL;
+    char *info_val = NULL;
+    MPI_Session shandle = MPI_SESSION_NULL;
+    MPI_Info sinfo = MPI_INFO_NULL;
+    MPI_Group pgroup = MPI_GROUP_NULL;
+
+    rc = MPI_Session_init(MPI_INFO_NULL, MPI_ERRORS_RETURN, &shandle);
+    if (rc != MPI_SUCCESS) {
+        MTestPrintfMsg(1, "Could not initialize session, bailing out\n");
+        return -1;
+    }
+
+    MPI_Session_get_num_psets(shandle, MPI_INFO_NULL, &n_psets);
+    for (i = 0, pset_name = NULL; i < n_psets; i++) {
+        psetlen = 0;
+        MPI_Session_get_nth_pset(shandle, MPI_INFO_NULL, i, &psetlen, NULL);
+        if (psetlen <= 0) {
+            printf("MPI_Session_get_nth_pset: returns psetlen = %d\n", psetlen);
+            errs++;
```

Comment:
For error reporting, I think we should use `fprintf(stderr, ...)` but not `MTestPrintfMsg ` nor `printf `.

Reply:
Actually, I am not sure what is the guideline. What is your reasoning?

Reply:
I think I'll just use `printf`. MTest utilities simply prints to `stdout`, so I think it is consistent to just print to `stdout`. `MTestPrintfMsg` observes `verbosity` setting, so I guess all tests that doesn't have a lot of error messages can simply `printf`.

Reply:
updated all tests to use `printf`

Reply:
I don't think we have any strict guideline for the error reporting. I just think it is a straightforward rule for programs to report any error on `stderr`. You are right that we are mixing the use of `printf` and `fprintf(stderr)` in many of our test programs. In fact, `MTestPrintErrorMsg` uses `printf` too. 

Anyway, I am OK if you prefer `printf` as long as it is consistent throughout the program.